### DISCUSSION TITLE
Convert Pandas DF from multiple CSVs into a sktime compatible format for classification

### DISCUSSION THREAD

**Discussion by TannerCypret** [alan-turing-institute/sktime#7220] [Q&A] [answered]
### Discussed in https://github.com/sktime/sktime/discussions/7216

<div type='discussions-op-text'>

<sup>Originally posted by **TannerCypret** October  2, 2024</sup>
Let's say I time series for 100 different aircraft containing latitude, longitude, altitude, and airspeed in individual CSVs each containing over a thousand rows of this data as a time series. What would be the best way to read them in so that I can pass them as an argument to TimeSeriesForestClassifier? I can loop through them and read them in as Pandas dataframes. I'm struggling to take the pandas dataframes and combine them all correctly.  I've been looking at the format of the provided data sets like Arrow Head. The examples show that each time in the time series corresponds to a single value, such as the shape of the arrow head. I have 4 or possibly more values if I want to include them. Can this be done with this classifier? If so, how are more data points supported? Would it be as a common or space separated list? 

For example the first arrow head time series values looks like this:

0     -1.963009
1     -1.957825
2     -1.956145
3     -1.938289
4     -1.896657
         ...   
I would want multiple values such as this:
0      x,y,z,speed
1      x,y,z,speed
2      x,y,z,speed
3      x,y,z,speed
4      x,y,z,speed
         ...   

Each of my time series are in a different CSV. Is there a pandas or sktime command that could help combine and format them?
</div>

**Comment by fkiraly** [comment, marked as answer]
(changed to discussion as this seems more like a question).

If you want to convert the data to `pd-multiindex` format, a quick way would be:

* write a loop to load all files into individual `pd.DataFrame`
* then apply `pandas.concat` with the `keys` argument - this creates a row-`MultiIndex`, ensure you name all series differently via `keys` (e.g., `range(N)`).

This should be in the `pd-multiindex` format, you can check with `check_is_mtype` from the `datatypes` module.
